lib/core/models/process_ora.dart

```
@@ -0,0 +1,33 @@
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+import 'package:archive/archive.dart';
+import 'package:image/image.dart' as img;
+
+class ProcessOra {
+  Future<List<ui.Image>> processOraFile(Archive archive) async {
+    List<ui.Image> layers = [];
+
+    for (var file in archive) {
+      if (file.isFile &&
+          (file.name.endsWith('.png') || file.name.endsWith('.jpg')) ||
+          file.name.endsWith('.ora')) {
+        img.Image? decodedImage = img.decodeImage(file.content as List<int>);
```

If `decodeImage fails,  this might silently skip it. More explicit error handling or logging here could be beneficial.